docs/guides/src/main/server/fips.adoc

```diff
@@ -0,0 +1,229 @@
+requirement as they are longer than 14 characters.
+
+* RSA keys of 1024 bits do not work (2048 is the minimum). This applies for keys used by Keycloak realm itself (Realm keys from the `Keys` tab in the admin console), but also client keys and IDP keys
+
+* HMAC SHA-XXX keys must be at least 112 bits (or 14 characters long). For example if you use OIDC clients with the client authentication `Signed Jwt with Client Secret` (or `client-secret-jwt` in
+the OIDC notation), then your client secrets should be at least 14 characters long. Note that for good security, it is recommended to use client secrets generated by Keycloak server, which
+always match this requirement.
+
+== Other restrictions
+
+In order to have SAML working, there is a need to have `XMLDSig` security provider to be available in your security providers.
+In order to have Kerberos working, there is a need to have `SunJGSS` security provider available. In FIPS enabled RHEL 9 in OpenJDK 17.0.6, these
+security providers are not by default in the `java.security`, which means that they effectively cannot work.
+
+To have SAML working, you can manually add the provider into `JAVA_HOME/conf/security/java.security` into the list fips providers. For example add the line like:
+
+```
+fips.provider.7=XMLDSig
+```
+
+Adding this security provider should be fine as in fact it is FIPS compliant and likely will be added by default in the future OpenJDK 17 micro version.
+Details are in the https://bugzilla.redhat.com/show_bug.cgi?id=1940064[bugzilla].
+
+NOTE: It is recommended to look at `JAVA_HOME/conf/security/java.security` and check all configured providers here and make sure that the number matches. In other words, `fips.provider.7`
+assumes that there are already 6 providers configured with prefix like `fips.provider.N` in this file.
+
+If you don't want to edit your `java.security` file inside java itself, the option is to create custom java security file (for example named `kc.java.security`) and add just the single
+property above for adding XMLDSig provider into that file. Then start your Keycloak server with this property file attached:
+
+```
+-Djava.security.properties=/location/to/your/file/kc.java.security
+```
+
+For Kerberos/SPNEGO, the security provider `SunJGSS` is not yet fully FIPS compliant. Hence it is not recommended to add it to your list of security providers
+if you want to be FIPS compliant. The `KERBEROS` feature is disabled by default in Keycloak when it is executed on FIPS platform and when security provider is not
+available. Details are in the https://bugzilla.redhat.com/show_bug.cgi?id=2051628[bugzilla].
+
+== Run the CLI on the FIPS host
+
+In case you want to run Client Registration CLI (`kcreg.sh|bat` script) or Admin CLI (`kcadm.sh|bat` script), it is needed
+that CLI will also use the BouncyCastle FIPS dependencies instead of plain BouncyCastle dependencies. To achieve this, you may copy the
```

Comment:
```suggestion
 the CLI must also use the BouncyCastle FIPS dependencies instead of plain BouncyCastle dependencies. To achieve this, you may copy the
```